### ISSUE TITLE
How can I use mongorestore with MupX since mongodb is inside docker container?

### ISSUE THREAD

**Issue by sferoze** [arunoda/meteor-up-legacy#816]
I used mongodump to backup my database since I want to move it from being hosted at compose.io to being hosted locally in the server itself using mupx. 

Once I setup the app and have it running, how can I restore the mongodump? I am using mupx, and when I ssh into the server I see that mongodb is inside a docker container.

What are the steps needed to use mongorestore given that I can copy the mongodump files from my local pc to the server.

**Comment by MasterJames** [comment]
I added full functionality for Mongo dump and restore to a PR which was never merged along with 40+ others.
https://github.com/arunoda/meteor-up/pull/736

What I learned from that is that the docker container mounts an external folder for Mongo and so you can see those files easily enough.
Doing it properly would take ssh or you can use docker exec like my PR.
I'll try to find the real folder and repost that (or add it here in an edit).

The install-mongodb.sh show the mount point 
https://github.com/arunoda/meteor-up/blob/mupx/scripts/linux/install-mongodb.sh

```
/var/lib/mongodb 
```

Is the local folder where the database data is stored and mounted in the Mongo docker container.

**Comment by sferoze** [comment]
@MasterJames 

I am still confused. 

To clarify here are the steps I understand so far:

1) Use `scp` command to copy the mongodump folder from my local pc to the server
2) SSH into the server

Then from this point where I am logged into the server and am in the same directory as the dump folder, what docker command am I supposed to run?

**Comment by MasterJames** [comment]
The easiest way I thought would be to use my Pull Request instead (probably not an easy task without cloning etc.).
Otherwise I can say the docker command is executed here like this...
https://github.com/MasterJames/meteor-up/blob/mupx/lib/actions.js#L481

```
docker exec -t mongodb mongorestore dumpfile
```

but you need a properly dumped dump file via mongodump --out dumpfile
https://github.com/MasterJames/meteor-up/blob/mupx/lib/actions.js#L414

Otherwise maybe you can just scp over /var/lib/mongodb from your source as I pointed to in the previous comments above. I've not tried that. Easy to backup the opposite way. I'm sure as well.
The problem with that is your suppose to use the cli tools mongodump and mongorestore.

https://docs.mongodb.org/manual/reference/program/mongodump/
https://docs.mongodb.org/manual/reference/program/mongorestore/

You should understand the issues detailed in those links. No default overwrite for one (use --drop).

I've shown you how to use docker exec if it's on a container you need to run a command (like dump/restore), but the link for more info is here...
https://docs.docker.com/engine/reference/commandline/exec/

I also though why not get the DB imported & running on your local dev & deployment system and then just deploy it from there? but I'm sure you have your reasons.

**Comment by sferoze** [comment]
Hi @MasterJames 

```
"I also though why not get the DB imported & running on your local dev & deployment system and then just deploy it from there? but I'm sure you have your reason"
```

I don't see how this would help. If I get the app running locally, and import the data into my local mongodb, when I deploy with mupx, it starts a whole new DB. Unless I am missing something, Mupx does not copy over local mongodb data to production server. 

So I need to setup a new server with my app and mongodb. Then mongorestore the data from the mongodb instance I had hosted on compose.io

**Comment by MasterJames** [comment]
It will transfer and run Mongo for you if you select the option in the settings.
More specifically it will put it on the target machine but mount that folder into a docker container running mongodb. 
https://github.com/arunoda/meteor-up/blob/mupx/templates/linux/start.sh

**Comment by sferoze** [comment]
@MasterJames which option do I set? Because if I set `"setupMongo": true` all that does it setup mongodb on the server. It does not copy over local database data to production. 

Is there another option for this? Typically you would NOT want to copy over the local DB to production. So I am assuming most people do not use this option.

**Comment by MasterJames** [comment]
Right it will leave it the second time so not to overwrite. Run mupx setup again to ovrewrite.  I think?  Or you could delete the folder off the target manually or use a new instance which I suppose is the expected routine.

https://github.com/arunoda/meteor-up/blob/mupx/scripts/linux/install-mongodb.sh#L17

If you ssh into your target and delete the folder /var/lib/mongodb 

Or you could scp the folder over too.

mupx setup

**Comment by sferoze** [comment]
@MasterJames 

I just want to let you know that mupx does not copy over your local mongodb instance to production. Even upon the first time using mupx setup. 

It never copies any local database data to production. 

I will have to do this manually.

**Comment by MasterJames** [comment]
Oh good to know best of luck and happy computing.

**Comment by sferoze** [comment]
I figured out how to do it

Here are the instructions

1) Copy dump folder to server

`scp -r /local_path/to/dump_folder root@111.222.33.4:/remote/path`

2) SSH into server

`ssh root@111.222.33.4`

3) Copy from root of server to inside docker container

`docker cp dump_folder mongodb:/dump_folder`

4) Go into mongodb docker container

`docker exec -it mongodb bash`

5) check if copied folder exists

`ls` (you should see dump_folder, if you named it the same folder as in this example)

6) use mongorestore

`mongorestore --drop -d AppName dump_folder`

**Comment by dburles** [comment]
@sferoze that should really be somewhere in the mupx docs! :)

**Comment by MasterJames** [comment]
Mupx doesn't do it. Only my PR does and it's documented.
Anyway since the Mongo docker contain uses the volume argument to mount a local file system you can simple use rcp or sshfs to transfer either way directly from the Ubuntu 14+ target. These require shutdown or restart and are not acceptable for a running production server. Then you want to run proper Mongo restore and backup with oplog options etc.

**Comment by dburles** [comment]
Edit: Just realised there's a new version in development :)

**Comment by MasterJames** [comment]
Yes at kadirahq it was suggested I port the PR and so not a priority as after I realized it's just sitting on the file system mounted into the Mongo docker container and easy to copy etc. 
Anyway I'm considering my own tool which might be considered useful by others but I think if ever it will be specific to AWS etc.

**Comment by zhukovsp** [comment]
@sferoze Thank you so much!

**Comment by krishaamer** [comment]
This would be very useful. Any update?

**Comment by alvara** [comment]
@sferoze Thank you! That was exactly what I was looking for.
